### ISSUE TITLE
Cloudwatch log queries not working after upgrading to Grafana 8.0

### ISSUE THREAD

**Issue by awsiv** [grafana/grafana#35448]
**What happened**:

Cloudwatch logs queries stopped working after upgrade to Grafana 8.0

Haven't tested in a new installation though

**What you expected to happen**:
Nothing mentioned in the changelogs about it.. so I expected it to work OOTB

**How to reproduce it (as minimally and precisely as possible)**:
1. Configure Cloudwatch datasource with "AWS SDK Default"
2. Go to Explore and do a simple CW log query on a log group (with data)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0
- Data source type & version: Cloudwatch 
- OS Grafana is installed on: Docker "grafana:latest" on fargate
- User OS & Browser: Windows/Brave
- Grafana plugins: Cloudwatch
- Others:

**Comment by csgergorubint** [comment]
Experiencing the same issue, Metrics queries are working just fine, running any log query will POST to /query, and return with 200, with the following response data:

```
{
  "results": {
    "A": {
      "frames": [
        {
          "schema": {
            "name": "A",
            "refId": "A",
            "meta": {
              "custom": {
                "channelName": "<uuid removed>"
              }
            },
            "fields": []
          },
          "data": {
            "values": []
          }
        }
      ]
    }
  }
}
``` 

**Environment:**

- [grafana helm chart](https://github.com/grafana/helm-charts/tree/main/charts/grafana) 6.12.1
- values.yaml excerpt:
```
serviceAccount:
  annotations:
    eks.amazonaws.com/role-arn: "<arn of the role>"

env:
  AWS_REGION: "eu-central-1"
  AWS_ROLE_ARN: "<arn of the role>"
  AWS_WEB_IDENTITY_TOKEN_FILE: "/var/run/secrets/eks.amazonaws.com/serviceaccount/token"
  

datasources:
 datasources.yaml:
   - name: CloudWatch
     type: cloudwatch
     access: proxy
     jsonData:
       authType: "default"
       defaultRegion: "default"
```
- IAM policy attached to the role
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "",
            "Effect": "Allow",
            "Action": [
                "tag:GetResources",
                "logs:StopQuery",
                "logs:StartQuery",
                "logs:GetQueryResults",
                "logs:GetLogGroupFields",
                "logs:GetLogEvents",
                "logs:DescribeLogGroups",
                "ec2:DescribeTags",
                "ec2:DescribeRegions",
                "ec2:DescribeInstances",
                "cloudwatch:ListMetrics",
                "cloudwatch:GetMetricStatistics",
                "cloudwatch:GetMetricData",
                "cloudwatch:DescribeAlarmsForMetric",
                "cloudwatch:DescribeAlarms",
                "cloudwatch:DescribeAlarmHistory"
            ],
            "Resource": "*"
        }
    ]
}
```

**Comment by hakro** [comment]
Hello,

I'm facing this issue as well. Cloudwatch logs stopped working after upgrading to 8.0 (including 8.0.2)

It's not an AWS permissions issue, I tried with Full Admin access and it didn't help.

Nothing much happens in the logs : 

`t=2021-06-14T15:25:53+0000 lvl=info msg="Initialized channel handler" logger=live channel=plugin/cloudwatch/3d52179a-3026-419f-965f-830df696f86d address=plugin/cloudwatch/3d52179a-3026-419f-965f-830df696f86d
`

**Comment by mattparkes** [comment]
This is also broken for us. We've tried `8.0.0`, `8.0.1` and `8.0.2`, as well as `Full Admin`. We get the same `200` response as @csgergorubint. 

**Environment:**

**Grafana version:** 8.0.x
**Data source type & version:** Cloudwatch
**OS Grafana is installed on:** Docker "`grafana/grafana:8.0.x`" on EKS (Amazon Linux 2)
**User OS & Browser:** Chrome on Mac OS
**Grafana plugins:** `
grafana-piechart-panel,raintank-worldping-app,jdbranham-diagram-panel,goshposh-metaqueries-datasource,vonage-status-panel,petrslavotinek-carpetplot-panel,grafana-polystat-panel,snuids-radar-panel,yesoreyeram-boomtheme-panel,digrich-bubblechart-panel,sbueringer-consul-datasource,agenty-flowcharting-panel,citilogics-geoloop-panel,michaeldmoore-multistat-panel,corpglory-progresslist-panel,marcuscalidus-svg-panel,fetzerch-sunandmoon-datasource,snuids-trafficlights-panel,https://github.com/mtanda/grafana-sumologic-datasource/releases/download/1.0.11/grafana-sumologic-datasource-1.0.11.zip;sumologic-datasource,grafana-image-renderer
`

**Comment by dgdevops** [comment]
Same issue here, also tried v8.0.2 but it did not bring any improvement for me either.

**Comment by aocenas** [comment]
Hey everyone, thanks for reporting. We already have a fix and it should be in 8.0.3 patch.

**Comment by pavelnemirovsky** [comment]
@aocenas version 8.0.3 didn't build due to some test failures https://github.com/grafana/grafana/pull/35746

**Comment by pavelnemirovsky** [comment]
@aocenas thanks for patching,I have installed 8.0.3 but it still returns an empty response. 
please advise.

**Comment by hristaki** [comment]
I can confirm too that the v8.0.3 is not resolving the problem.

**Comment by aocenas** [comment]
When building from source I can reproduce the issue in 8.0.2 tag but I get logs as expected in 8.0.3.

@pavelnemirovsky, @hristaki can you comment how do you run/install grafana (docker, rpm package or something else)?
Also can you confirm that you see the correct version?
![Screenshot from 2021-06-21 09-35-56](https://user-images.githubusercontent.com/1014802/122724519-4c1de900-d274-11eb-8536-8c9b422d609a.png)

When you don't see any logs, do you see the logs panel like this?
![Screenshot from 2021-06-21 09-37-03](https://user-images.githubusercontent.com/1014802/122724584-5fc94f80-d274-11eb-993e-6fe122649478.png)

**Comment by yongzhang** [comment]
not sure my issue is related to this one: https://github.com/grafana/grafana/issues/35974

**Comment by williamium** [comment]
@aocenas I'm also seeing the problem persist with 8.0.3, the same version as in your screenshot. I'm using the RPM.

**Comment by hakro** [comment]
After upgrading to 8.0.3, it works fine on my side.
I use the official Docker image.

**Comment by hristaki** [comment]
I am installing Grafana direct on Ubuntu following the documentation here: https://grafana.com/grafana/download?pg=get&plcmt=selfmanaged-box1-cta1

```
wget https://dl.grafana.com/oss/release/grafana_8.0.3_amd64.deb
sudo dpkg -i grafana_8.0.3_amd64.deb
```

**Comment by hastarin** [comment]
I am using an image based on the Docker image (grafana/grafana:8.0.3-ubuntu) and I'm also seeing this issue.

**Comment by Limess** [comment]
We're also seeing this issue on 8.0.3. Logs worked fine on 7.x.x, now we get a 200 response and no logs being shown, nothing in the Grafana logs. We're using the same IAM permissions as above

Running Grafana in Docker on ECS (official Grafana image as base without any real changse), behind an nginx proxy and an ALB.

**Comment by sam-burrell** [comment]
Also seeing this on 8.0.3. 
Same behaviour as above.

**Comment by aocenas** [comment]
So seems like there may be some other issue apart from the one that got fixed in 8.0.3 as now it works for some (and for me). So will need a bit more information, so if you comment please respond to these if you can:

- Is there any error message in UI, browser console or Grafana logs?
- What auth type are you using? If SDK default can you try Access & secret key and see if that works?
- Do you get a normal response just without data or something else?
- Is the problem just with logs or also with cloud watch metrics queries?

**Comment by awsiv** [comment]
I can confirm that CW logs are working in 8.0.3:

My setup:

- release package: grafana/grafana:8.0.3 (from docker hub)
- Deployed to fargate behind loadbalancer
- Using SDK default for auth

Haven''t tested further, as far a the the scope of this issue is concerned - it looks fixed to me

**Comment by csgergorubint** [comment]
The issue is resolved for me with helm chart version 6.13.2, with the same configuration as in my previous [comment](https://github.com/grafana/grafana/issues/35448#issuecomment-859404867).

**Comment by hristaki** [comment]
> So seems like there may be some other issue apart from the one that got fixed in 8.0.3 as now it works for some (and for me). So will need a bit more information, so if you comment please respond to these if you can:
> 
> * Is there any error message in UI, browser console or Grafana logs?

I see some messages in the console:
![image](https://user-images.githubusercontent.com/4948421/123040114-8235a700-d3f3-11eb-8018-8b0e4dbae15d.png)

> * What auth type are you using? If SDK default can you try Access & secret key and see if that works?
Access & secret key 

> * Do you get a normal response just without data or something else?
for the metric:  i see the message "no data"
for the logs:  the spinning wheel is just spinning the whole time

> * Is the problem just with logs or also with cloud watch metrics queries?
Both

**Comment by andrew-nye** [comment]
Good day to you. I can confirm that it is still not working for myself in 8.0.3. With regards to the requested information by @aocenas it is exactly the same as provided by @hristaki.

**Comment by williamium** [comment]
> - Is there any error message in UI, browser console or Grafana logs?

No errors in the Grafana log, the only error I see in the browser console is the same websocket error as @hristaki, and there's no error in the UI - you just get a spinner that never stops.

> - What auth type are you using? If SDK default can you try Access & secret key and see if that works?

SDK Default with an IAM role

> - Do you get a normal response just without data or something else?

I still see the same [response](https://github.com/grafana/grafana/issues/35448#issuecomment-859404867) as reported by @csgergorubint.

> - Is the problem just with logs or also with cloud watch metrics queries?

Well maybe to confuse things further, I'm seeing something a little different to some others... Metrics work for me, as they have always done. The issue is specifically for logs.

I doubt it's helpful, but we also have the AWS X-Ray datasource, and that works as expected too. It also uses SDK default with an IAM role for authentication.

**Comment by daniel-bray-sonalake** [comment]
@aocenas we're also seeing an error here.

Visibly it looks like the request never completes, but I can see from the inspector that it does, and successfully.  I also can't spot any errors in the cloudwatch logs (we're set to debug) or in the chrome console (Version 91.0.4472.114 (Official Build) (64-bit)).

I have two instances of grafana querying the same cloudwatch (we're testing if the upgrade is safe).

- `Grafana v7.5.4 (615c153b3a)` works fine
  -  runs in docker using image`grafana/grafana:7.5.4`
- `Grafana v8.0.3 (cae5c5e46b)` was created by starting the v8.0.3 against a snapshot of the v7.5.4 version's DB.
  -  runs in docker using image`grafana/grafana:8.0.3`

  
The test was to run the same query on data sources configured the same way:

![image](https://user-images.githubusercontent.com/18031246/123131108-4aefe600-d445-11eb-8a0c-7dd5f37fc4f6.png)

What I see is that the  v7.5.4 version runs to completion (it finds nothing, but I expect that), but the v8.0.3 just runs forever without any error:

![image](https://user-images.githubusercontent.com/18031246/123131162-580cd500-d445-11eb-8f6b-c113efb849ee.png)


The call to `/api/ds/query` completes in about 500ms, with a 200 OK response

I get a response like this in the inspector
```
{
  "request": {
    "method": "POST",
    "url": "api/ds/query",
    "data": {
      "from": "1624460640111",
      "to": "1624464240112",
      "queries": [
        {
          "intervalMs": 1,
          "maxDataPoints": 1,
          "datasourceId": 3,
          "queryString": "fields @timestamp, @message\n| filter @message like /find nothing here/",
          "refId": "A",
          "logGroupNames": [
            "API-Gateway-Execution-Logs_b34x7dice7/orders-prd"
          ],
          "statsGroups": [],
          "region": "eu-west-1",
          "type": "liveLogAction"
        }
      ]
    },
    "hideFromInspector": false
  },
  "response": {
    "results": {
      "A": {
        "frames": [
          {
            "schema": {
              "name": "A",
              "refId": "A",
              "meta": {
                "custom": {
                  "channelName": "b6efc70f-7c16-4caa-947e-5988483c35e0"
                }
              },
              "fields": []
            },
            "data": {
              "values": []
            }
          }
        ],
        "refId": "A"
      }
    }
  }
}
```


And here are the logs I get. You can see it's for the same channel name, and there are no errors: 

```
t=2021-06-23T16:04:00+0000 lvl=dbug msg="Processing metrics query" logger=http.server query="&{data:map[datasourceId:3 intervalMs:1 logGroupNames:[API-Gateway-Execution-Logs_b34x7dice7/orders-prd] maxDataPoints:1 queryString:fields @timestamp, @message\n| filter @message like /find nothing here/ refId:A region:eu-west-1 statsGroups:[] type:liveLogAction]}"
t=2021-06-23T16:04:00+0000 lvl=dbug msg="Querying for data source via SQL store" logger=datasources id=3 orgId=1
t=2021-06-23T16:04:00+0000 lvl=dbug msg="Client subscribed" logger=live user=2 client=49b471ea-2b16-4870-add6-decc4dd3d556 channel=1/plugin/cloudwatch/b6efc70f-7c16-4caa-947e-5988483c35e0
t=2021-06-23T16:04:00+0000 lvl=dbug msg="Client wants to subscribe" logger=live user=2 client=49b471ea-2b16-4870-add6-decc4dd3d556 channel=1/plugin/cloudwatch/b6efc70f-7c16-4caa-947e-5988483c35e0
t=2021-06-23T16:04:00+0000 lvl=info msg="Initialized channel handler" logger=live channel=plugin/cloudwatch/b6efc70f-7c16-4caa-947e-5988483c35e0 address=plugin/cloudwatch/b6efc70f-7c16-4caa-947e-5988483c35e0
```

**Comment by daniel-bray-sonalake** [comment]
A comment on this https://github.com/grafana/grafana/issues/35448#issuecomment-866983897

I've noticed that it works ok when I restart the service, but after the first few calls - with no errors - it starts to fail

**Comment by daniel-bray-sonalake** [comment]
What's also odd is that I can create a duplicate data source that works, i.e.

1. Same plugin (cloudwatch)
2. Same config ("Assume Role ARN" using the same role)
3. Same default region

The only difference is that this new data source was created after the upgrade and so wasn't used in any existing dashboards or alerts

**Comment by yongzhang** [comment]
> What's also odd is that I can create a duplicate data source that works, i.e.
> 
> 1. Same plugin (cloudwatch)
> 2. Same config ("Assume Role ARN" using the same role)
> 3. Same default region
> 
> The only difference is that this new data source was created after the upgrade and so wasn't used in any existing dashboards or alerts

I created a new datasource as you mentioned, but still not working when querying cloudwatch logs.

**Comment by daniel-bray-sonalake** [comment]
I spoke too soon. It looks like if I recreate the datasource it works for a while, before it eventually fails too.

This is only happening for cloudwatch logs queries, not for metrics queries (they work fine)

No errors in the logs:


```	
t=2021-06-28T08:36:21+0000 lvl=dbug msg="Processing metrics query" logger=http.server query="&{data:map[datasourceId:70 intervalMs:1 logGroupNames:[JccFeedmasterLogs] maxDataPoints:1 queryString:fields @timestamp, @message refId:A region:eu-west-1 statsGroups:[] type:liveLogAction]}"
t=2021-06-28T08:36:21+0000 lvl=dbug msg="Querying for data source via SQL store" logger=datasources id=70 orgId=1
t=2021-06-28T08:36:21+0000 lvl=info msg="Initialized channel handler" logger=live channel=plugin/cloudwatch/40aec637-2dcc-4def-96e6-1d1951b9ae2b address=plugin/cloudwatch/40aec637-2dcc-4def-96e6-1d1951b9ae2b
t=2021-06-28T08:36:21+0000 lvl=dbug msg="Client wants to subscribe" logger=live user=2 client=1512ebb4-fc63-45b7-b3a6-b5ecda2517e8 channel=1/plugin/cloudwatch/40aec637-2dcc-4def-96e6-1d1951b9ae2bShow context
t=2021-06-28T08:36:21+0000 lvl=dbug msg="Client subscribed" logger=live user=2 client=1512ebb4-fc63-45b7-b3a6-b5ecda2517e8 channel=1/plugin/cloudwatch/40aec637-2dcc-4def-96e6-1d1951b9ae2b
```

No errors on screen, no results shown, it just loops forever.

Like I said, this was working ok for about 30 minutes or so after I deleted/re-created the data source.

**Comment by anybodysguest** [comment]
This issues doesn’t appear to be fixed. My scenario:

Using 8.0.3 image
Deployed in Fargate stack
CloudWatch metrics work fine
CloudWatch logs however, never complete (spins forever).
There doesn’t appear to be anything that would diagnose issues in the logs.
I have all the required permissions in my IAM role. Even tried specifying “cloudwatch.*” and that didn’t make a difference.
I tried going into go into developers tools (firefox) while the query is running (I’m trying to explore the logs) and it looks like it’s throwing a 400 error. See screen shot(my Grafana domain name has been redacted).
Reverted back to 7.5.9, no issues.

<img width="1913" alt="Screen Shot 2021-06-27 at 10 50 44 AM" src="https://user-images.githubusercontent.com/79932609/123663794-99a5d300-d7eb-11eb-9788-e0ba7a212422.png">

**Comment by daniel-bray-sonalake** [comment]
@aocenas, can this issue be reopened? I think enough people are seeing this issue that we can say there are still parts of this that need to be fixed.

**Comment by JackMorgan9** [comment]
> So seems like there may be some other issue apart from the one that got fixed in 8.0.3 as now it works for some (and for me). So will need a bit more information, so if you comment please respond to these if you can:
> 
> * Is there any error message in UI, browser console or Grafana logs?
> * What auth type are you using? If SDK default can you try Access & secret key and see if that works?
> * Do you get a normal response just without data or something else?
> * Is the problem just with logs or also with cloud watch metrics queries?

Using 8.0.3

- No error messages are available in logs or UI
- Using SDK, just tried AK/SK, Same issue.
- We get a continuous load
- Metrics are working fine, just appears to be logs as the issue

Same issue as others where if we restart grafana, loads for one or 2 queries then stops working again.

**Comment by caveman280** [comment]
Hi all,

I've been looking to bump our Grafana installation to v8.0.3 and have also experienced this too - there were a couple of things missed out from the changelog which caught us by surprise.

**Our Stack:**
For reference, I'll include our stack so it's clear where we're at. We have multiple Linux instances (which have Nginx and Grafana installed; Nginx terminates mTLS connections and proxies them to Grafana installed on the instance), which sit inside an Auto Scaling Group behind a Load Balancer. We use the CloudWatch plugin (authenticating by the `AWS SDK Default` provider, where we assume a role in a given region) and also experience no result back from CloudWatch Logs (well, it spins and loads forever).

1. **Grafana Live** is now enabled by default (Reference: [Configure Grafana Live](https://grafana.com/docs/grafana/latest/live/configure-grafana-live/)).
    - This means that websockets are now enabled by default, which requires any proxy servers sitting in front of Grafana are now required to allow websockets.
       -  @anybodysguest, @hristaki  - that looks like your issue. You see the `400` status code response from the `/api/live/ws`? That means your Nginx isn't letting them through - see: (https://grafana.com/docs/grafana/latest/live/configure-grafana-live/#websocket-and-proxies).
    - Sadly, `https://grafana.com/tutorials/run-grafana-behind-a-proxy/` still doesn't include mention any about websocket handling and is easier to find using Google!
1. **Grafana Live** doesn't support High Availability (reference: [Configure Grafana Live HA setup](https://grafana.com/docs/grafana/latest/live/live-ha-setup/#configure-grafana-live-ha-setup)).
    - With Grafana Live not supporting High Availability, this makes the feature unusable for us. Grafana is used as our 'Single Source of Truth' meaning that the majority of our metrics are visualised in Grafana. We cannot have a situation where Grafana is inaccessible at any time of the day, so a single Grafana instance is not acceptable.
    - [Set up Grafana for high availability](https://grafana.com/docs/grafana/latest/administration/set-up-for-high-availability/) has no warning about breaking Grafana live if enabled.
    - Luckily, we can disable Grafana Live and restore our HA infrastructure (reference: [Configuration](https://grafana.com/docs/grafana/latest/administration/configuration/#max_connections)), _bringing us onto our next headache..._
1. **CloudWatch Logs** returns no results when `max_connections` for Grafana Live is set to `0`.
    - So, as we can't use Grafana Live due to HA, we've disabled it. However, the CloudWatch plugin doesn't take this into account. I performed the following debugging steps (as mentioned earlier in this issue, logging (even debug logging) doesn't yield much useful information):
       - Start the query in edit / view mode of the table in Grafana.
       - Observe in AWS CloudTrail that the Grafana instance assumes the role defined in the settings of Grafana's CloudWatch Plugin.
       - Look for all events from that assumed role user in CloudTrail and observe that we have an API call for `DescribeLogGroups`, `DescribeRegions`, `StartQuery` **but you never see a `GetQueryResults`**.
    - I've been trying to work out why that might be, and I believe it's due to the following PR:
       - Because even if Grafana Live is disabled.... it still sends it over the websocket.... which doesn't exist.
       -  #33654, specifically: https://github.com/grafana/grafana/pull/33654/files#diff-5195d59bff5abb37dceeb2ba37e2e0acb2f73f861fc83e320a72f6e7b2784f18L131-R131
```diff
-      if (config.featureToggles.live) {
-         dataQueryResponses.push(this.handleLiveLogQueries(logQueries, options));
-      } else {
-        dataQueryResponses.push(this.handleLogQueries(logQueries, options));
-      }
+      dataQueryResponses.push(this.handleLiveLogQueries(logQueries, options));
+      //  dataQueryResponses.push(this.handleLogQueries(logQueries, options));
```

Feel free to tell me to raise a new issue for anything I've mentioned!

Thank you for all the hard work and continual improvement, I'm really looking forward to the day we can rollout Grafana v8!

**Comment by aocenas** [comment]
@caveman280 thank you very much for your comprehensive investigation. This really helps a lot also with the PR. You are right and I am going to merge the PR shortly. As you say this may not solve all the issues but after your PR is merged they can check if disabling the live feature will help and afterwards figure out the proxy setting.

Probably no need to create separate issues just yet. The HA + Live issue is tracked and worked on and the other ones are better tracked here as already lots of people are looking at this issue.

**Comment by aocenas** [comment]
For people having issues, there should be another patch release this week with #36358. This will allow you to disable [Live feature](https://grafana.com/docs/grafana/latest/live/configure-grafana-live/) and allow CloudWatch logs to work without websockets. So if you still have problems afterwards you can try disabling the Live feature by setting:
```
[live]
max_connections = 0
```
in your Grafana config. If querying works after that, it is highly probable that the issue is in your proxy configuration which does not handle websocket connections, in which case you can take a look [here](https://grafana.com/docs/grafana/latest/live/configure-grafana-live/#websocket-and-proxies).

See [comment](https://github.com/grafana/grafana/issues/35448#issuecomment-872224175) for more details

**Comment by daniel-bray-sonalake** [comment]
Thanks


> it is highly probable that the issue is in your proxy configuration which does not handle websocket connections

I'll check again once this is merged to a release, but please note we're deploying to AWS ECS with an Application LB acting as our proxy, and these support websocket connections by default.

**Comment by anybodysguest** [comment]
> Thanks
> 
> > it is highly probable that the issue is in your proxy configuration which does not handle websocket connections
> 
> I'll check again once this is merged to a release, but please note we're deploying to AWS ECS with an Application LB acting as our proxy, and these support websocket connections by default.

Same here.  We're using ECS/Fargate with an ALB.

**Comment by mycrEEpy** [comment]
Had the same (or a similar?) issue with `/api/live/ws` throwing `403` status codes and Cloudwatch Logs not showing anything.
The only thing we fixed in our Grafana configuration was to set the `domain` and `root_url` to their proper values as described in [here](https://grafana.com/docs/grafana/latest/live/configure-grafana-live/#request-origin-check). After that it started working again.

For reference: our Grafana setup is proxied by an NLB & Nginx Ingress running on EKS.
The Nginx Ingress rule is configured for WebSockets with `nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"` and `nginx.ingress.kubernetes.io/proxy-send-timeout: "3600"`

**Comment by davidnewhall** [comment]
`/api/live/ws` gives me a 201. I see pings and responses happening in Chrome dev console. Web sockets appear to be working. The only proxy is an AWS ALB. The logs panels still do not populate and throw an error. Disabling web sockets as explained by @aocenas fixes the problem. Something still seems wrong though, and I'm not sure what
![Screen Shot 2021-07-12 at 1 33 54 PM](https://user-images.githubusercontent.com/2402929/125352209-d27a9780-e315-11eb-92b2-8e7f5763cc28.png)
 else to troubleshoot.
Grafana 8.0.4 and 8.0.5 exhibit the same behavior. Enterprise running in Fargate.

**Comment by aocenas** [comment]
@davidnewhall To clarify, that `Cannot read property 'meta' of undefined`, is an error that you get after you disable web sockets or do you see that with web sockets enabled and it goes away when you disable them?

**Comment by davidnewhall** [comment]
I began getting that error after upgrading from 7.5.x to 8.0. Once 8.0.5 came out I tried max_connections = 0 and the error goes away.

**Comment by davidnewhall** [comment]
I noticed this in your commit message on the PR that closed this issue.
> It's likely deployments might currently disable Grafana Live, given that it doesn't support HA yet

We're running Grafana Enterprise in Fargate. I guess that means HA and doesn't work? hm. Am I receiving the expected error?

**Comment by aocenas** [comment]
@davidnewhall As far as I know Grafana Live not supporting HA means that features needing websockets will not work properly in HA setup. Like the live updates of dashboards may work for some people but not for others, because the websockets are connected to only one instance and there is no mechanism yet to share the state between instances so it will depend on who is connected to which instance.

The error you get is a bit weird and seems like some other front end issue that may or may not be directly related to the websockets problem, at least I did not see that while reproducing this.

**Comment by FZambia** [comment]
@caveman280 hello, many thanks for a detailed comment - I tried to address some of your thoughts in related pull requests. Maybe you will have any suggestions.

**Comment by carlosgranada** [comment]
Hi 👋 we are having exactly this same issue with Grafana cloud. The logs panels were working perfectly fine and suddenly they stopped working. CloudWatch Logs never return and spins forever. CloudWatch Metrics are ok and working, the connection to the data source
<img width="1600" alt="Screen Shot 2021-07-31 at 9 55 52 AM" src="https://user-images.githubusercontent.com/923923/127742184-f82e79ab-58b6-43d7-9db2-8b03d2a7f38d.png">
 is working.

**Comment by anybodysguest** [comment]
It sounds like we're back where we started from.  8.x doesn't work for me, only 7.x does.  Back to using the latest of 7.x (7.5.10).  I'm having the same problem as @carlosgranada.  I'm not using Grafana Cloud though, i'm running this in AWS Fargate.

**Comment by hastarin** [comment]
I thought I'd mention that on a self hosted (Docker) version of Grafana I've recently set it to restart the service daily as log queries seem to stop working after some period of time. Sadly I have too many other things on to track down further.

**Comment by anybodysguest** [comment]
Are there any plans to re-open this issue to deal with the CW issue.  The problem still exists an it's keeping me from moving to  8.x.  I use a lot of CloudWatch logs in my dashboards.  CloudWatch metrics work fine however, it's just the logs that I and others are having this issue of the queries never completing.

**Comment by awsiv** [comment]
@anybodysguest did you try this suggestion btw?

https://github.com/grafana/grafana/issues/35448#issuecomment-875536941

**Comment by davidnewhall** [comment]
This solved my problem on two Grafana installations (running in Fargate). It also required Grafana 8.0.5+ (8.0.5 has a different bug, so definitely jump to 8.0.6+).
```
[live]
max_connections = 0
```

I also tried sticky sessions on the ALB but that did not help. Good luck!

**Comment by anybodysguest** [comment]
@awsiv 
Tried setting max_connections to 0.  It didn't change anything (e.g. CW log query still hanging).  I set the connections by passing the following parameter to my Docker container:   GF_LIVE_MAX_CONNECTIONS=0

**Comment by daniel-bray-sonalake** [comment]
The above workaround works fine for me now, with:

- Grafana v8.1.2 (103f8fa094)
- Running in AWS Fargate with 2 tasks running in the service, behind a stateless ALB
- After adding these environment variables 
```
# disable websockets, fargate doesn't like them
- GF_LIVE_MAX_CONNECTIONS=0 
# enable the new alert features
- GF_FEATURE_TOGGLES_ENABLE=ngalert 
```

**Comment by FZambia** [comment]
Should be solved by https://github.com/grafana/grafana/pull/39231 – will be part of Grafana 8.2.0 (and also 8.2.0 betas).

**Comment by aocenas** [comment]
As mentioned by @FZambia it should be solved here #39231. Means cloud watch will not use websockets to get results whether Live feature is enabled or not.
